group-coordinator/src/main/java/org/apache/kafka/coordinator/group/GroupCoordinatorService.java

```diff
@@ -846,6 +861,152 @@ public CompletableFuture<DeleteGroupsResponseData.DeletableGroupResultCollection
             (accumulator, newResults) -> newResults.forEach(result -> accumulator.add(result.duplicate())));
     }
 
+    List<String> deleteCandidateGroupIds(
```

Comment:
Should it be private? If we use it from tests, let's add `// package-private for testing` please.